jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitAuthManager.kt

```diff
@@ -30,11 +34,68 @@ class DefaultToolkitAuthManager : ToolkitAuthManager, PersistentStateComponent<T
             )
         }
 
-        factoryConnections.toList()
+        factoryConnections.toMutableSet()
+    }
+
+    init {
+        // initial load then subscribe to bus for future changes
```

Comment:
Would adding this as a listener work instead?

Reply:
yes, the listener does this, but we don't want to worry about the order of which service gets initialized first